pkg/core/managers/apis/mesh/mesh_manager_test.go

```diff
@@ -296,33 +245,18 @@ var _ = Describe("Mesh Manager", func() {
 		It("should not allow to change CA when mTLS is enabled", func() {
 			// given
 			meshName := "mesh-1"
-			resKey := model.ResourceKey{
-				Name: meshName,
-			}
 
 			// when
-			mesh := core_mesh.MeshResource{
-				Spec: &mesh_proto.Mesh{
-					Mtls: &mesh_proto.Mesh_Mtls{
-						EnabledBackend: "builtin-1",
-						Backends: []*mesh_proto.CertificateAuthorityBackend{
-							{
-								Name: "builtin-1",
-								Type: "builtin",
-							},
-						},
-					},
-				},
-			}
-			err := resManager.Create(context.Background(), &mesh, store.CreateBy(resKey))
+			mesh := samples.MeshMTLSBuilder().WithName(meshName)
+			err := mesh.Create(resManager)
 
 			// then
 			Expect(err).ToNot(HaveOccurred())
 
 			// when trying to change CA
 			mesh.Spec.Mtls.Backends[0].Name = "builtin-2"
 			mesh.Spec.Mtls.EnabledBackend = "builtin-2"
-			err = resManager.Update(context.Background(), &mesh)
+			err = resManager.Update(context.Background(), mesh.Build())
```

Comment:
`Build` seems easy to accidentally skip since we apparently have direct access to the resource, maybe just be honest and call it `Validate`?

Or maybe there's an interface where one first calls `MeshTLSBuilderFrom()` to create a new builder from an existing resource and then has to call `meshModifier.Build()`

Reply:
see https://github.com/kumahq/kuma/pull/5123#discussion_r1000612826
Now `Build` makes more sense since the resource is hidden